src/session.rs

```diff
@@ -375,34 +375,55 @@ impl Session {
                 return;
             }
         };
-        let clock_rate = match media.get_params(header.payload_type) {
-            Some(v) => v.spec().clock_rate,
+        let (clock_rate, is_rtx) = match media.get_params(header.payload_type) {
+            Some(v) => (v.spec().clock_rate, v.resend() == Some(header.payload_type)),
             None => {
                 trace!("No codec params for {:?}", header.payload_type);
                 return;
             }
         };
 
-        // Figure out which SSRC the repairs header points out. This is here because of borrow
-        // checker ordering.
-        let ssrc_repairs = header
-            .ext_vals
-            .rid_repair
-            .and_then(|repairs| media.ssrc_rx_for_rid(repairs));
-
         let source = media.get_or_create_source_rx(ssrc);
 
         let mut media_need_check_source = false;
-        if let Some(rid) = header.ext_vals.rid {
+
+        // Obtain the rid supplied in the header, if any
+        let supplied_rid = if is_rtx {
+            header.ext_vals.rid_repair
+        } else {
+            header.ext_vals.rid
+        };
+
+        // Determine the source rid, and update if one was supplied
+        let rid = if let Some(rid) = supplied_rid {
             if source.set_rid(rid) {
                 media_need_check_source = true;
             }
-        }
-        if let Some(repairs) = ssrc_repairs {
-            if source.set_repairs(repairs) {
-                media_need_check_source = true;
+            supplied_rid
+        } else {
+            source.rid()
+        };
+
+        // Gymnastics to appease the borrow checker.
+        let source = if is_rtx {
+            // For RTX, we will look for the associated repair ssrc using the last RID seen
+            // on our source.
+            let primary_ssrc = media.find_primary_ssrc_for_rid(rid, ssrc);
+
+            let source = media.get_or_create_source_rx(ssrc);
+
+            if let Some(primary_ssrc) = primary_ssrc {
+                if source.set_repairs(primary_ssrc) {
+                    media_need_check_source = true;
+                }
+            } else {
+                trace!("Ignoring RTX packet because the we could not find an associated ssrc");
```

Comment:
There's a problem here: you just added the rtx source, but didn't call .set_repairs on it, which means that future calls to .is_rtx() on it will be false, and it may be chosen as the primary source.  I think this may happen if the first packing coming in is an RTX packet.

So perhaps it's better to do something like this:

```
let Some(primary_ssrc) = primary_ssrc else {
  trace!("Ignoring RTX packet because the we could not find an associated
  return;
}
let source = media.get_or_create_source_rx(ssrc);
if source.set_repairs(primary_ssrc) {
  media_need_check_source = true;
}
source
```